diff --git a/cmake/Cuda.cmake b/cmake/Cuda.cmake
index 0e755d05..c12948c4 100644
--- a/cmake/Cuda.cmake
+++ b/cmake/Cuda.cmake
@@ -1,6 +1,6 @@
 # Known NVIDIA GPU achitectures Caffe can be compiled for.
 # This list will be used for CUDA_ARCH_NAME = All option
-set(Caffe_known_gpu_archs "30 35 50 52 60 61 70 75")
+set(Caffe_known_gpu_archs "35 37 50 52 53 60 61 62 70 75 80 86")
 
 ################################################################################################
 # A function for automatic detection of GPUs installed  (if autodetection is enabled)
